### PR TITLE
random LLSEC fixes

### PR DESCRIPTION
**Pull request by kkrentz**
* READMEs describing `llsec` and `noncoresec`
* It is no longer necessary to use a static network-wide security level in LLSEC drivers
* Support for using a custom FRAMER in `noncoresec`
* If LLSEC is off, `packetbuf.h` currently defines "pseudo-attritubtes". This can cause hard-to-find bugs if LLSEC is configured improperly. Therefore, I removed this "feature".

Apart from that, I also overcame various limitations of `noncoresec`, but that is going to be part of another pull request - see https://github.com/kkrentz/contiki/tree/master/core/net/llsec/adaptivesec

### CODE REVIEW

File: core/net/mac/tsch/tsch-security.h [hunk 1]
```diff
@@ -39,22 +39,36 @@
...
-#endif /* TSCH_SECURITY_ENABLED */
-#if TSCH_SECURITY_ENABLED && LLSEC802154_CONF_USES_FRAME_COUNTER
-#error TSCH_SECURITY_ENABLED set but LLSEC802154_CONF_USES_FRAME_COUNTER set
-#endif /* TSCH_SECURITY_ENABLED */
+
+#ifdef TSCH_SECURITY_CONF_LEVEL
+#define TSCH_SECURITY_LEVEL TSCH_SECURITY_CONF_LEVEL
+#else /* TSCH_SECURITY_CONF_LEVEL */
+#define TSCH_SECURITY_LEVEL FRAME802154_SECURITY_LEVEL_NONE
+#endif /* TSCH_SECURITY_CONF_LEVEL */
```
Comment (simonduq) [inline core/net/mac/tsch/tsch-security.h]: I don't think we need such a flag at all: TSCH security levels (and keys and key indexes) are defined in https://github.com/contiki-os/contiki/blob/master/core/net/mac/tsch/tsch-security.h

We used `LLSEC802154_CONF_SECURITY_LEVEL` so far only in a binary way (sec enabled or not), I guess `LLSEC802154_ENABLED` is now perfectly suited for the task.

Comment (kkrentz) [inline core/net/mac/tsch/tsch-security.h]: TSCH_SECURITY_LEVEL is gone

**Comment by simonduq** [comment]
Looks great to me! :+1: (except I still don't understand why link-layer security should sit above MAC, see https://github.com/contiki-os/contiki/pull/1383)

**Comment by kkrentz** [comment]
I have added a fix for Issue #1537

**Comment by simonduq** [comment]
Let me add a timeout and schedule for merging.

**Comment by simonduq** [comment]
merging after timeout

**Comment by tonnenpinguin** [comment]
Sorry to bring this up after merging, but I'm getting a build error caused by `LLSEC802154_CONF_ENABLED` not being defined.

Shouldn't [this line](https://github.com/contiki-os/contiki/blob/master/core/net/mac/rdc.h#L54) check for `LLSEC802154_ENABLED` rather than `LLSEC802154_CONF_ENABLED`?

**Comment by kkrentz** [comment]
Somehow, my compiler does not throw any error, while it should. Can you check whether #1625 solves your problem.

**Comment by tonnenpinguin** [comment]
For me that does fix the issue, yes! (Also, IMHO it does make sense to stick with common practice)

**Pull request merged**